internal/broker/broker.go

```diff
@@ -636,6 +631,44 @@ func (b *Broker) handleIsAuthenticated(ctx context.Context, session *session, au
 	return AuthGranted, userInfoMessage{UserInfo: authInfo.UserInfo}
 }
 
+// userNameIsAllowed checks whether the user's username is allowed to access the machine.
+func (b *Broker) userNameIsAllowed(userName string) bool {
+	// The user is allowed to login if:
+	// - ALL users are allowed
+	// - the user's name is in the list of allowed_users
+	// - the user is the owner of the machine and OWNER is in the allowed_users list
+	if b.cfg.userConfig.allUsersAllowed {
+		return true
+	} else if _, ok := b.cfg.userConfig.allowedUsers[userName]; ok {
+		return true
+	} else if b.cfg.userConfig.ownerAllowed {
+		// If owner is undefined, then the first user to login is considered the owner
+		if b.cfg.userConfig.firstUserBecomesOwner || b.cfg.userConfig.owner == userName {
+			return true
+		}
+	}
+	return false
```

Comment:
We tend to avoid the `else` when it's not needed (because of the `return` in the `if` block)

```suggestion
	if b.cfg.userConfig.allUsersAllowed {
		return true
	}

	if _, ok := b.cfg.userConfig.allowedUsers[userName]; ok {
		return true
	}

	if b.cfg.userConfig.ownerAllowed &&
		(b.cfg.userConfig.firstUserBecomesOwner || b.cfg.userConfig.owner == userName) {
		return true
	}

	return false
}
```
```